[docs] Add DeferNode and ConditionNode to query plan docs (#3002)
apollographql/federation: docs/source/query-plans.mdx

| Node | Description |
|------|-------------|
| [`Fetch`](#fetch-node) | Tells the router to execute a particular operation on a particular subgraph. |
| [`Parallel`](#parallel-node) | Tells the router that the node's immediate children can be executed in parallel. |
| [`Sequence`](#sequence-node) | Tells the router that the node's immediate children must be executed serially in the order listed. |
| [`Flatten`](#flatten-node) | Tells the router to merge the data returned by this node's child `Fetch` node with data previously returned in the current `Sequence`. |

> | [`Defer`](#defer-node) | Tells the router about one or more `@defer` blocks at the same nested level. It contains a primary block and an array of deffered blocks. |

Reviewer: ```suggestion
| [`Defer`](#defer-node) | Tells the router about one or more blocks of `@defer`ed fields at the same level of nesting. The node contains a primary block and an array of deferred blocks. |
```

Now that you've learned about each query plan node, take another look at the example query plan in [Example graph](#example-graph) to see how these nodes work together in a complete query plan.

### Defer node

A `Defer` node corresponds to one or more `@defer`s at the same level of nesting in the query plan.

The node contains a _primary block_ and an array of _deferred blocks_. The primary block represents the part of the query that isn't deferred. Each deferred block corresponds to the one deferred part of the query.

> Read more about how `@defer` operatates at [Apollo Router support for @defer](/router/executing-operations/defer-support/).

Reviewer: ```suggestion
Read more about how `@defer` works in the [Apollo Router support for @defer article](/router/executing-operations/defer-support/).
```

```graphql
QueryPlan {
  Defer {
    Primary {
      Fetch(...) {}
    }, [
      Deferred(...) {
        Flatten(...) {
          Fetch(...) {}
        }
      }
    ]
  }
}
```

### Condition node

> A `Condition` node splits a query plan into an if-else branch. `Condition` nodes are commonly used for `@skip` and `@include` directives so the query plan to select different nodes based on the provided runtime variables.

Reviewer: ```suggestion
A `Condition` node splits a query plan into an if-else branch. `Condition` nodes are commonly used for `@skip` and `@include` directives so the query plan can select different nodes based on the provided runtime variables.
```

```graphql

> QueryPlan {
>   Sequence {
>     Fetch(...) {}
>     Include(...) {
>       Flatten(...) {
>         Fetch(...) { }
>       }
>     }
>   }
> }
> ```
> ```graphql
> QueryPlan {
>   Sequence {
>     Fetch(...) {}
>     Skip(...) {
>       Flatten(...) {
>         Fetch(...) { }
>       }
>     }
>   }
> }

Reviewer: I don't see a `Condition` node in the current example. Is this correct?:

```suggestion
QueryPlan {
  Condition(if: $condition) {
    Then {
      Sequence {
        Fetch(...) {}
        Include(...) {
          Flatten(...) {
            Fetch(...) { }
          }
       }
    }  Else {
       Sequence {
         Fetch(...) {}
         Skip(...) {
           Flatten(...) {
             Fetch(...) { }
          }
       }
    }
 }
```

Author: Yes, I will update the doc header. We renamed the node in a recent Federation version which I generated the example query plans from

    - Note that you must [register your federated graph](./managed-federation/setup/) with GraphOS to view query plans in the Explorer.
- As direct output from the `@apollo/gateway` library (see below)

### Defer node

> A `Defer` node corresponds to one or more `@defer` at the same level of "nestedness" in the planned query.
> 
> It contains a "primary block" and an array of "deferred blocks". The "primary block represents the part of the query that is not deferred (so the part of the query up until we reach the @defer(s) this handle), while each "deferred block" correspond to the deferred part of one of the `@defer` handle by the node.
> 
> Read more about how `@defer` operatates at [Apollo Router support for @defer](/router/executing-operations/defer-support/).
> 

Reviewer: ```suggestion

A `Defer` node corresponds to one or more `@defer`s at the same level of nesting in the query plan.

The node contains a _primary block_ and an array of _deferred blocks_. The primary block represents the part of the query that isn't deferred. Each deferred block corresponds to the one deferred part of the query.

Read more about how `@defer` operatates at [Apollo Router support for @defer](/router/executing-operations/defer-support/).

```

### Condition node

> A `Condition` node is used to split a plan into an if-else branch. This is commonly used for `@skip` and `@include` to have a query plan that can select different nodes based on the provided runtime variables.

Reviewer: ```suggestion

A `Condition` node splits a query plan into an if-else branch. `Condition` nodes are commonly used for `@skip` and `@include` directives so the query plan to select different nodes based on the provided runtime variables.
```

> 

Reviewer: It looks like these need to move up to the previous H2, right under the `### Flatten node` section, so they are not part of `## Viewing query plans`

Additionally, could you please add examples of what the node looks like, as is done in the other node sections? E.g.  for [Flatten node](https://www.apollographql.com/docs/federation/query-plans/#flatten-node)
